src/index.ts

```diff
@@ -12,25 +12,100 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import {OctokitAPIs} from './github';
+import {OctokitAPIs, GitHub} from './github';
+import {ReleaseType} from './releasers';
+import {ReleasePR} from './release-pr';
+import {ChangelogSection} from './conventional-commits';
 
-export {ReleaseCandidate, ReleasePR, ReleasePROptions} from './release-pr';
+export {ReleaseCandidate, ReleasePR} from './release-pr';
 
-// Configuration options shared by Release PRs and
-// GitHub releases:
-export interface SharedOptions {
+// Used by GitHub: Factory and Ctor
+interface GitHubOptions {
+  defaultBranch?: string;
+  token?: string;
+  apiUrl?: string;
+  octokitAPIs?: OctokitAPIs;
+}
+
+// Used by GitHubRelease and ReleasePR: Factory and Ctor
+interface ReleaserOptions {
   label?: string;
-  repoUrl: string;
+}
+
+// Used by GitHubRelease: Factory and Ctor
+interface GitHubReleaseOptions extends ReleaserOptions {
+  changelogPath?: string;
+  draft?: boolean;
+}
+
+// Used by ReleasePR: Factory and Ctor
+interface ReleasePROptions extends ReleaserOptions {
   path?: string;
   packageName?: string;
+  bumpMinorPreMajor?: boolean;
+  releaseAs?: string;
+  snapshot?: boolean;
   monorepoTags?: boolean;
-  token?: string;
-  apiUrl: string;
-  octokitAPIs?: OctokitAPIs;
+  fork?: boolean;
+  changelogSections?: ChangelogSection[];
+  // only for Ruby: TODO replace with generic bootstrap option
+  lastPackageVersion?: string;
+  // for Ruby: TODO refactor to find version.rb like Python finds version.py
+  // and then remove this property
+  versionFile?: string;
+}
+
+// GitHub Ctor options
+export interface GitHubCtorOptions extends GitHubOptions {
+  owner: string;
+  repo: string;
+}
+
+// Used by GitHubRelease and ReleasePR Ctor
```

Comment:
nit: might spell this out constructor, had to Google the shorthand.

Reply:
will do. I had to look it up too not that long ago (finally got tired of seeing it everywhere on SO and not really knowing what it mean). I come from python where `init` is both short and accurate :-)